addons/sale/models/sale_order.py

```diff
@@ -771,6 +793,15 @@ def _compute_is_expired(self):
                 and order.validity_date < today
             )
 
+    @api.depends('order_line.qty_delivered')
+    def _compute_show_deliver_button(self):
```

Comment:
I would do an override in purchase_stock with self.show_deliver_button = False
To avoid override xml everywhere to hide the button